### PR TITLE
[TIMOB-23908] (5_5_X) Fixed APS environment entitlements. Removed ExternalAcc…

### PR DESCRIPTION
**Pull request by cb1kenobi**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-23908

Fixed APS environment entitlements. Removed ExternalAccessory.framework when building an app that does not contain a watch app.

### CODE REVIEW

File: iphone/cli/commands/_build.js [hunk 1]
```diff
@@ -3403,18 +3410,16 @@ iOSBuilder.prototype.writeEntitlementsPlist = function writeEntitlementsPlist()
 	// if we have a provisioning profile, make sure some entitlement settings are correct set
 	if (pp) {
 		// attempt to customize it by reading provisioning profile
+		if (!plist.hasOwnProperty('application-identifier')) {
+			plist['application-identifier'] = pp.appPrefix + '.' + this.tiapp.id;
+		}
+		plist['aps-environment'] = this.target === 'dist-appstore' || this.target === 'dist-adhoc' ? 'production' : 'development';
```
Comment (hansemannn) [inline iphone/cli/commands/_build.js]: Should we really set the `aps-environment` key although mit might not be inside the provisioning profile? I liked the below check.

File: iphone/cli/commands/_build.js [hunk 2]
```diff
@@ -2626,7 +2632,7 @@ iOSBuilder.prototype.createXcodeProject = function createXcodeProject(next) {
 						child.comment = '"' + appName + '.app"';
 					} else if (child.comment === 'Titanium.entitlements') {
 						child.comment = '"' + appName + '.entitlements"';
-					} else if (child.comment === 'LaunchScreen.storyboard' && appc.version.lt(this.xcodeEnv.version, '7.0.0')) {
+					} else if ((child.comment === 'LaunchScreen.storyboard' && appc.version.lt(this.xcodeEnv.version, '7.0.0')) || child.comment === 'ExternalAccessory.framework') {
```
Comment (hansemannn) [inline iphone/cli/commands/_build.js]: Not sure if that the `ExternalAccessory` change is related to this ticket. Let's not mix too many changes together if possible.

Comment (cb1kenobi) [inline iphone/cli/commands/_build.js]: It was causing the Wireless Accessory Configuration capability to be enabled. I'm simply removing it if there's no watch app.

File: iphone/cli/commands/_build.js (hunk 1 above)
Comment (cb1kenobi) [inline iphone/cli/commands/_build.js]: OK, so we could do this: 

```javascript
if (pp.apsEnvironment) {
    plist['aps-environment'] = this.target === 'dist-appstore' || this.target === 'dist-adhoc' ? 'production' : 'development';
}
```

If they use APS, then we set it, but we set it to values based on the target, not the value in the provisioning profile.

**Comment by hansemannn** [comment]
Approved!

**Pull request merged**